src/common/models.py

```diff
@@ -20,6 +20,34 @@ class ProcessingState(Enum):
     AWAITING = auto()
     SUBMITTED = auto()
 
+    def previous_states(self) -> tuple[ProcessingState, ...]:
+        """Possible previous states in processing state machine"""
+        match self:
+            case ProcessingState.AWAITING:
+                return ()
+            case ProcessingState.SUBMITTED:
+                return (ProcessingState.AWAITING,)
+            case ProcessingState.FAILURE_NONRETRYABLE:
+                return (ProcessingState.SUBMITTED,)
+            case ProcessingState.FAILURE_RETRYABLE:
+                return (ProcessingState.SUBMITTED,)
+            case ProcessingState.SUCCESS:
+                return (
+                    ProcessingState.SUBMITTED,
+                    ProcessingState.FAILURE_NONRETRYABLE,
+                    ProcessingState.FAILURE_RETRYABLE,
+                )
+
+    def migrate_logs_to_state(self) -> ProcessingState | None:
+        """Whether to copy logs for this state to another state"""
+        match self:
+            case (
+                ProcessingState.SUBMITTED
```

Comment:
I don't think we'll want to keep "submitted" logs, but added this as a no-change refactor

ref: https://github.com/NASA-IMPACT/hls-nextgen-orchestration/pull/2#discussion_r2392864445